contracts/SnappAuction.sol

```diff
@@ -240,6 +244,7 @@ contract SnappAuction is SnappBase {
                 auctionIndex == MAX_UINT || auctionIndex < 2 || auctions[auctionIndex - 2].appliedAccountStateIndex != 0,
                 "Too many pending auctions"
             );
+        indexTransitionBlock[auctionIndex] = block.number;    
```

Comment:
This method is only invoked when the first order of a new batch comes in. What happens if someone wants to submit a solution without anyone else having submitted an order.

Reply:
The way we have implemented the auctionIndex increase is that the index can only increase if there is a new order coming in.

Reply:
Right so if you are trying to solve batch 12 before batch 13 is opened, you will access `indexTransitionBlockp[12]` in the other method which has not been created yet. I guess that goes somewhat along the same lines as your comment about `blockhash` only working for the last 256 blocks.